main/adapters/repository/protocol.go

```diff
@@ -160,6 +160,36 @@ func (p *ExtendedProtocol) StoreSignature(tx TransactionCtx, uid uuid.UUID, sign
 	return tx.Commit()
 }
 
+func (p *ExtendedProtocol) StoreExternalIdentity(ctx context.Context, extId ent.ExternalIdentity) (err error) {
+	// store public key raw bytes
+	extId.PublicKey, err = p.PublicKeyPEMToBytes(extId.PublicKey)
```

Comment:
I don't understand this line. It seems to me it basically assigns extID.PublicKey to extID.PublicKey but with a conversion. But since both are []byte this would mean the format of extID.PublicKey is not determined. Maybe I am not getting something here.

Reply:
you are correct, this is bad design. i am aware of that, but have not found a better solution for this yet.

Reply:
A simple (maybe a bit naive solution) might be to keep one variable for each format: PublicKeyRaw and PublicKeyPEM or similar.